src/imfilter.jl

```diff
@@ -840,12 +840,33 @@ function _imfilter_fft!(r::AbstractCPU{FFT},
     out
 end
 
-function filtfft(A, krn)
+function planned_fft(A::AbstractArray{T,N},
+                    kernel::Tuple{AbstractArray,Vararg{AbstractArray}},
+                    border::BorderSpecAny=Pad(:replicate)) where {T,N}
+    bord = border(kernel, A, Algorithm.FFT())
+    _A = padarray(T, A, bord)
+    p1 = plan_rfft(_A)
+    kern = samedims(_A, kernelconv(kernel...))
+    krn = FFTView(zeros(eltype(kern), map(length, axes(_A))))
+    p2 = plan_rfft(krn)
+    B = p1 * _A
+    B .*= conj!(p2 * krn)
+    p3 = plan_irfft(B, length(axes(_A, 1)))
+    return Algorithm.FFT(p1, p2, p3)
```

Comment:
Can any of these actual operations be replaced with just constructing things based on dims etc.?